src/modules/simulator/simulator_mavlink.cpp

```diff
@@ -367,9 +381,8 @@ void Simulator::handle_message_hil_sensor(const mavlink_message_t *msg)
 			last_integration_us = 0;
 		}
 
-		float ibatt = -1.0f; // no current sensor in simulation
-
-		battery_percentage = math::max(battery_percentage, _param_bat_min_pct.get() / 100.f);
+		battery_percentage = 1.0f - consumed_mah / _battery.capacity();
```

Comment:
Works, with the default capacity of -1 you end up with `1 - (0 / -1) = 1` 👍 
```suggestion
		battery_percentage = 1.0f - (consumed_mah / _battery.capacity());
```